zebedee-cms/src/main/java/com/github/onsdigital/zebedee/util/SlackNotification.java

```diff
@@ -117,52 +208,139 @@ public static void send(String message, String slackToken, String slackChannel,
+
+    public static void scheduledPublishFailure(Collection c) {
+        if (c == null) {
+            // not enough info to be able to notify anything useful.
+            logDebug("scheduledPublishFailure collection is null").log();
+            return;
+        } else if (c.getDescription() == null) {
+            // not enough info to be able to notify anything useful.
+            logDebug("scheduledPublishFailure collection description is null").log();
+            return;
+        }
+
+        collectionAlarm(c, "Scheduled collection failed to publish");
+    }
 
     /**
      * Send a slack message containing collection publication information
      *
-     * @param publishedCollection
+     * @param collection
      */
-    public static void publishNotification(PublishedCollection publishedCollection, boolean publishComplete) {
-        if (publishedCollection == null) {
+    public static void publishNotification(CollectionBase collection, CollectionStage stage, StageStatus status) {
+        if (collection == null) {
             logWarn("failed to send publish slack notification as published collection was null").log();
             return;
         }
 
-        // default to unsuccessful message.
-        String slackMessage = "Collection " + publishedCollection.getName() + " did not publish";
-
-        if (publishComplete) {
-            Result result = publishedCollection.publishResults.get(0);
-            StringBuilder msg = new StringBuilder("Published collection: " + publishedCollection.getName());
+        PostMessage pm = new PostMessage(slackUsername, slackPublishChannel, PostMessage.Emoji.ChartWithUpwardsTrend, "Unknown publish stage or status");
+        PostMessageAttachment attch = new PostMessageAttachment("", "", PostMessageAttachment.Color.Good);
+        pm.attachments.add(attch);
+        addCollectionAttachment(attch, collection);
 
+        boolean forceNewMessage = false;
 
-            if (publishedCollection.publishStartDate != null && publishedCollection.publishEndDate != null) {
-                msg.append(" publish time : " + format.format(publishedCollection.publishStartDate));
-
-                String timeTaken = String.format("%.2f", (publishedCollection.publishEndDate.getTime()
-                        - publishedCollection.publishStartDate.getTime()) / 1000.0);
-
-                msg.append(" time taken: " + timeTaken + " ");
-            }
-
-            result.transaction.uriInfos
-                    .stream()
-                    .filter(info -> info.uri.endsWith(DATA_JSON))
-                    .findFirst()
-                    .ifPresent(urlInfo -> {
-                        msg.append("Example Uri: http://www.ons.gov.uk")
-                                .append(urlInfo.uri.substring(0, urlInfo.uri.length() - (DATA_JSON).length()));
-                    });
-            slackMessage = msg.toString();
+        switch (stage) {
+            case PrePublish:
+                switch (status) {
```

Comment:
[Personal opinion] I don't like the nested switch here I think it adds a lot of noise. I think it would be cleaner to have the top level switch here and then have separate methods for the stages. E.g.

```
switch (stage) {
            case PrePublish:
                prePublishNotification(...)
                break;
            case Publish:
                publishNotification(...)
                break;
....
```